From 3e7d6cb1663f21e702f8a6125c5695b66573af0a Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 19 Dec 2024 20:03:42 +0100
Subject: [PATCH 774/841] fix(stm32mp2): remove stm32mp2_ddr_helpers for M33
 TDCID

Remove call and compilation of stm32mp2_ddr_helpers in BL31 for
M33 TDCID as self refresh of DDR is managed by M33 firmware.

This patch allows to reduce the BL31 size.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 394e92a86f1962d6395d50073c5582f763dadd44 fix(stm32mp2): remove stm32mp2_ddr_helpers for M33 TDCID]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I008aa308be14eec64876c2649b703286bb172c8e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/426137
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/platform.mk   | 2 ++
 plat/st/stm32mp2/stm32mp2_pm.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 778e95f10..0893fbc43 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -370,7 +370,9 @@ BL31_SOURCES			+=	plat/st/stm32mp2/bl31_plat_setup.c			\
 					plat/st/stm32mp2/stm32mp2_pm.c				\
 					plat/st/stm32mp2/stm32mp2_topology.c
 
+ifeq ($(STM32MP_M33_TDCID),0)
 BL31_SOURCES			+=	drivers/st/ddr/stm32mp2_ddr_helpers.c
+endif
 
 # Generic GIC v2
 include drivers/arm/gic/v2/gicv2.mk
diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index b534c1089..69db42056 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -962,8 +962,10 @@ static void __dead2 stm32_system_off(void)
 
 	dcsw_op_all(DCCISW);
 
+#if !STM32MP_M33_TDCID
 	/* Force DDR off */
 	ddr_sub_system_clk_off();
+#endif
 
 	/* Prevent interrupts from spuriously waking up this cpu */
 	stm32mp_gic_cpuif_disable();
-- 
2.47.2

